plugins/http_plugin/include/eosio/http_plugin/http_plugin.hpp

```diff
@@ -157,6 +158,26 @@ namespace eosio {
 
       error_info error;
    };
+
+   enum class http_params_types {
+      no_params_required = 0,
+      params_required = 1
+   };
+
+   template<typename T, http_params_types require_params>
```

Comment:
I am unconvinced that its appropriate to pass this in as a template parameter.  However, if there is good reason for that then the conditional should use `if constexpr` as the outer conditional with the body check inside that to clearly delineate where the compile time behavior change(s) are expected.

Reply:
The template enforce the type and can do compile time optimization. Good point to make it constexpr. Is that okay, if I continue with template?